app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java

```diff
@@ -443,10 +444,34 @@ public boolean onFling(final MotionEvent e1, final MotionEvent e2, final float v
             }
         });
         binding.swipeUp.setOnClickListener(v -> {
-            final Object tag = v.getTag();
-            if (tag instanceof CharSequence) {
-                Utils.openURL(context, tag.toString());
-            }
+			final Object tag = v.getTag();
+		    if (tag instanceof CharSequence) {
+				DialogInterface.OnClickListener dialogClickListener = new DialogInterface.OnClickListener() {
+		        @Override
+		        public void onClick(DialogInterface dialog, int which) {
+					switch (which){
+						case DialogInterface.BUTTON_POSITIVE:
+							//Yes button clicked
+		                    Utils.openURL(context, tag.toString());
+		                    break;
+
+		                case DialogInterface.BUTTON_NEGATIVE:
+		                    //No button clicked
+		                    dialog.dismiss();
+		                    break;
+
+		               	default:
+		               		dialog.dismiss();
+		               		break;
+		           	}
+		      	}
+		  		};
+
+		   		AlertDialog.Builder builder = new AlertDialog.Builder(context);
+		   		builder.setTitle(R.string.swipe_up_confirmation);
+		   		builder.setMessage(tag.toString()).setPositiveButton("Yes", dialogClickListener)
+		   			.setNegativeButton("No", dialogClickListener).show();
```

Comment:
You don't need to require the same listener on the negative button. In fact you can just use a null...

Also string resources for buttons (They already exist)

```suggestion
		   		AlertDialog.Builder builder = new AlertDialog.Builder(context);
		   		builder.setTitle(R.string.swipe_up_confirmation);
		   		builder.setMessage(tag.toString()).setPositiveButton(R.string.yes, (d, w) -> Utils.openURL(context, tag.toString()))
		   			.setNegativeButton(R.string.no, null).show();
```

Of course, if you're wary about whether the dialog will actually dismiss or not, you can apply a similar lambda function onto the negative button

Reply:
Thank you for the review and the pointers! I'll work on it. 
Also as for the intendations, it looked fine in my IDE, but I guess it changed after I pushed. I'll try to fix that too.